extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -345,3 +376,19 @@ function isPositionValid(anchorElement, position) {
     return false;
   });
 }
+
+
+/**
+ * Calculates the appropriate height for a full-width responsive ad of the
+ * given width.
+ * @param {!{width: number, height: number}} viewportSize
```

Comment:
Ideally this would be a typedef defined in viewport-impl.js that is used by getSize.  I leave it to you if you want to do that refactor...

Reply:
Actually just used the width and height directly since there are getWidth and getHeight methods.